lib/teamsnap.rb

```diff
@@ -131,15 +131,24 @@ def init(opts = {})
       apply_endpoints(self, collection) && true
     end
 
-    def run(via, href, args = {})
+    def run(via, href, args = {}, opts = {})
       resp = client.send(via, href, args)
 
       if resp.status == 200
-        Oj.load(resp.body)
-          .fetch(:collection)
+        collection = Oj.load(resp.body).fetch(:collection)
+        if opts[:backup_cache]
```

Comment:
I like having it overwrite the file each time though, because I would want changes in the API to be automatically reflected in that file each startup.. rather than having to manually clear the file. make sense? or see any major issues with that?

I think the overhead is going to be pretty low on this considering this will only be called on application init